modules/gdscript/gdscript.cpp

```diff
@@ -2080,20 +2080,26 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 	//notification is not virtual, it gets called at ALL levels just like in C.
 	Variant value = p_notification;
 	const Variant *args[1] = { &value };
+	const StringName &notification_str = GDScriptLanguage::get_singleton()->strings._notification;
 
-	List<GDScript *> pl;
-	GDScript *sptr = script.ptr();
-	while (sptr) {
-		if (p_reversed) {
-			pl.push_back(sptr);
+	static LocalVector<GDScript *> script_stack;
```

Comment:
This cannot be `static`. I'm guessing you removed it because of the advice against it. But by making it static, you still expose it to nested call problems. Please read [Aaron's comment](https://github.com/godotengine/godot/pull/94118#pullrequestreview-2651346500) for details.


```suggestion
	LocalVector<GDScript *> script_stack;
```